smarts/core/sumo_traffic_simulation.py

```diff
@@ -273,6 +273,10 @@ def _close_traci_and_pipes(self):
             self._traci_conn.close()
             self._traci_conn = None
 
+    def _remove_all_vehicles(self):
```

Comment:
This method name is slightly misleading since it is specifically remove vehicles that are not SUMO vehicles.

Reply:
Probably the `union` is misleading or not explicitly enough?

Reply:
Oh sorry, when I originally looked at this I was reviewing late at night and submitted it when I got up.

Is the MAD vehicle somehow included in this?

Reply:
Yes, including MAD vehicles.